File: baekjoon/단계별로 풀어보기/큐, 덱/18258_큐 2.py
# https://www.acmicpc.net/problem/18258

from collections import deque
from sys import stdin

input = stdin.readline
q = deque()
for _ in range(int(input())):
    s = input().rstrip()
    if s == 'pop':
        print(q.popleft() if q else -1)
    elif s == 'size':
        print(len(q))
    elif s == 'empty':
        print(0 if q else 1)
    elif s == 'front':
        print(q[0] if q else -1)
    elif s == 'back':
        print(q[-1] if q else -1)
    else:
        q.append(s.split()[-1])

File: C#/Baekjoon/큐, 덱/18258_큐 2.cs
//https://www.acmicpc.net/problem/18258

using System.Text;

using var sr = new StreamReader(new BufferedStream(Console.OpenStandardInput()));
using var sw = new StreamWriter(new BufferedStream(Console.OpenStandardOutput()));
var sb = new StringBuilder();
var q = new Queue<int>();
int n = int.Parse(sr.ReadLine());
string s, last = "";
while (n-- > 0)
{
    s = sr.ReadLine();
    if (s.Contains(' '))
    {
        last = s.Split()[1];
        q.Enqueue(int.Parse(last));
    }
    else if (s == "pop") sb.AppendLine(q.Count == 0 ? "-1" : $"{q.Dequeue()}");
    else if (s == "size") sb.AppendLine($"{q.Count}");
    else if (s == "empty") sb.AppendLine(q.Count == 0 ? "1" : "0");
    else if (s == "front") sb.AppendLine(q.Count == 0 ? "-1" : $"{q.Peek()}");
    else if (s == "back") sb.AppendLine(q.Count == 0 ? "-1" : last);
}
sw.Write(sb);